File: src/Servers/QueryReport/Entity/Structure/Redis/PeerRoomInfo.cs
using System;
using System.Collections.Generic;
using UniSpyLib.Database.DatabaseModel.MySql;

namespace QueryReport.Entity.Structure.Redis
{
    public class PeerRoomInfo
    {
        public static readonly List<string> GameSpyStandardKey =
            new List<string>
            {
                "groupid","hostname","numwaiting",
                "maxwaiting","maxplayers","numservers",
                "numplayers","password","numGames",
                "numplaying"
            };

        public Dictionary<string, string> KeyValue { get; protected set; }

        public DateTime UpdateTime;

        public PeerRoomInfo(Grouplist grouplist)
        {
            KeyValue = new Dictionary<string, string>();

            KeyValue.Add("groupid", grouplist.Groupid.ToString());
            KeyValue.Add("hostname", grouplist.Roomname);
            KeyValue.Add("numwaiting", "0");
            KeyValue.Add("maxwaiting", "200");
            KeyValue.Add("maxplayers", "200");
            KeyValue.Add("numservers", "0");
            KeyValue.Add("numplayers", "0");
            KeyValue.Add("password", "");
            KeyValue.Add("numGames", "0");
            KeyValue.Add("numplaying", "0");

            UpdateTime = DateTime.Now;

        }
        public PeerRoomInfo()
        {
            KeyValue = new Dictionary<string, string>();
        }
    }
}

File: src/frontends/gamespy/protocols/query_report/aggregates/peer_room_info.py
from pydantic import BaseModel, Field


class PeerRoomInfo(BaseModel):
    game_name: str
    group_id: int = Field(..., alias='groupid')
    room_name: str = Field(alias="hostname")
    number_of_waiting: int = Field(default=0, alias="numwaiting")
    max_waiting: int = Field(default=200, alias='maxwaiting')
    number_of_servers: int = Field(default=0, alias="numservers")
    number_of_players: int = Field(default=0, alias="numplayers")
    max_players: int = Field(default=200, alias="maxplayers")
    password: str = Field(default="", alias="password")
    number_of_games: int = Field(default=0, alias="numgames")
    number_of_playing: int = Field(default=0, alias="numplaying")

    def get_gamespy_dict(self) -> dict:
        """
        return a immutable dict
        """
        data = self.model_dump(mode="json")
        del data["game_name"]
        return data
